File: sys/frontend/user/mobile/android/app/src/main/kotlin/app/gengoka/domain/model/Challenge.kt
package app.gengoka.domain.model

data class Challenge(
    val id: String,
    val categoryId: String,
    val title: String,
    val description: String?,
    val charLimit: Int,
    val releaseDate: String?,
    val answerCount: Int
)

data class ChallengeWithCategory(
    val id: String,
    val categoryId: String,
    val title: String,
    val description: String?,
    val charLimit: Int,
    val releaseDate: String?,
    val answerCount: Int,
    val category: Category?,
    val isCompleted: Boolean = false,
    val userAnswer: Answer? = null
)

data class DailyChallenge(
    val challenge: Challenge,
    val categoryName: String,
    val isCompleted: Boolean,
    val userAnswer: Answer? = null
)

File: sys/frontend/user/web/src/types/challenge.ts
import type { Answer } from './answer';
import type { Category } from './category';

export interface Challenge {
  id: string;
  category_id: string;
  title: string;
  description: string | null;
  char_limit: number;
  release_date: string | null;
  answer_count: number;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface ChallengeWithCategory extends Challenge {
  category: Category;
}

export interface DailyChallengeResponse {
  challenge: Challenge;
  category_name: string;
  is_completed: boolean;
  user_answer: Answer | null;
}
